option optimize_for = SPEED;

package NQueryData;

///////////////////////////////////////
// interface data types
///////////////////////////////////////

/// key types

/// !!!Do not change names!!!

enum EKeyType {
    KT_QUERY_EXACT = 0;
    KT_QUERY_LOWERCASE = 2;
    KT_QUERY_SIMPLE = 8;
    KT_QUERY_STRONG = 11;
    KT_QUERY_DOPPEL = 1;
    KT_QUERY_DOPPEL_TOKEN = 7;
    KT_QUERY_DOPPEL_PAIR = 14;

    KT_USER_ID = 4;
    KT_USER_LOGIN = 5;
    KT_USER_LOGIN_HASH = 12;
    KT_USER_REGION = 16;
    KT_USER_REGION_IPREG = 6;
    KT_USER_IP_TYPE = 21;

    KT_DOCID       = 9;
    KT_SNIPDOCID   = 18;
    KT_CATEG  = 19;
    KT_SNIPCATEG = 3;
    KT_CATEG_URL = 22;
    KT_SNIPCATEG_URL = 23;
    KT_URL = 25;
    KT_URL_NO_CGI = 26;

    KT_SERP_TLD    = 17;
    KT_SERP_UIL    = 10;
    KT_SERP_DEVICE = 20;

    KT_STRUCTKEY = 15;
    KT_BINARYKEY = 13;

    KT_COUNT = 27;
}
